blog/_posts/2020-11-06-hybrid-cloud-with-inlets.md

```diff
@@ -0,0 +1,212 @@
+
+### Start the inlets-pro client
+
+Now that the server part of the tunnel is running, it is time to start the client in our private data center.
+Let's say we have a MySQL instance available with an internal IP address `10.1.0.50`, start the inlets-pro client:
+
+``` bash
+$ inlets-pro client --license-file ~/inlets-license --port 3306 --url wss://inlets.example.com:8123/connect --upstream 10.1.0.50 --token <your token> 
+2020/11/05 13:23:21 Welcome to inlets-pro! Client version 0.7.2
+2020/11/05 13:23:21 Licensed to: Johan Siebens <xxxx@gmail.com>, expires: xxx day(s)
+2020/11/05 13:23:21 Upstream server: 10.1.0.50, for ports: 3306
+inlets-pro client. Copyright Alex Ellis, OpenFaaS Ltd 2020
+INFO[2020/11/05 13:23:21] Connecting to proxy                           url="wss://inlets.example.com:8123/connect"
+```
+
+Perfect! Now the client made the connection, port 3306 of the server pod in our public cloud is accepting connection and will tunnel traffic to the MySQL instance.
+
+### Create a MySQL service
+
+When we deploy WordPress, we could configure it to connect directly to the inlets-pro server pod, but it is better to create Kubernetes Service:
+
+``` yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: mysql
+  labels:
+    app: mysql
+spec:
+  ports:
+    - name: mysql
+      port: 3306
+      targetPort: 3306
+  selector:
+    app: inlets-pro-server
+```
+
+The set of Pods targeted by this Service is determined by the same selector as the previous service, but this time it is a service of type ClusterIP, making it only accessible from inside the cluster.
+
+### Deploy WordPress
+
+The only thing left for our example is deploying a WordPress instance, connecting to the MySQL database via the inlets-pro tunnel:
+
+``` yaml
+apiVersion: apps/v1
+metadata:
+  name: wordpress
+  labels:
+    app: wordpress
+spec:
+  selector:
+    matchLabels:
+      app: wordpress
+  template:
+    metadata:
+      labels:
+        app: wordpress
+    spec:
+      containers:
+      - image: wordpress
+        name: wordpress
+        env:
+        - name: WORDPRESS_DB_HOST
+          value: mysql
+        ports:
+        - containerPort: 80
+          name: wordpress
+```
+
+> note: this WordPress is not production-ready as it is missing the required volumes for the content
+
+Mission accomplished! Our WordPress application, running in a public cloud environments is using the MySQL server located in the private data center.
+
+## Wrapping up
+
+This tutorial gives us a short introduction on how inlets PRO can help us to build a hybrid cloud between existing servers and public cloud.
+As a cheaper, easier alternative to a data-center uplink or managed product like AWS Direct Connect or Azure Express Route it is a very lightweight, but powerful, tool to bring your on-prem services to a cloud workload.
+
+The WordPress example is rather a simple use case, but with the same techniques, other, more complex, architectures are possible too.
```

Comment:
`The WordPress example is rather a simple use case` <- we could be more positive with:

For the example we chose WordPress, but the same technique can be applied to any other applications that use TCP traffic.